sdk/core/azure-core/azure/core/pipeline/transport/base.py

```diff
@@ -81,6 +83,37 @@ def _urljoin(base_url, stub_url):
     return parsed.geturl()
 
 
+class _HTTPSerializer(HTTPConnection):
+    """Hacking the stdlib HTTPConnection to serialize HTTP request as strings.
+    """
+    def __init__(self, *args, **kwargs):
+        self.skip_host = kwargs.pop("skip_host", True)
+        self.skip_accept_encoding = kwargs.pop("skip_accept_encoding", True)
+        self.buffer = b''
+        kwargs.setdefault("host", "fakehost")
+        super(_HTTPSerializer, self).__init__(*args, **kwargs)
+
+    def putheader(self, header, *values):
+        if self.skip_host and header == "Host":
+            return
+        if self.skip_accept_encoding and header == "Accept-Encoding":
+            return
+        super(_HTTPSerializer, self).putheader(header, *values)
+
+    def send(self, data):
+        self.buffer += data
+
+def _serialize_request(http_request):
+    serializer = _HTTPSerializer()
```

Comment:
Is `_HTTPSerializer` expected to be used by others elsewhere? If not why do `skip_host` and `skip_accept_encoding` need to be explicitly configurable?

Reply:
Habits, doesn't need to :)

Reply:
Fixed